pkg/rancher-desktop/preload/extensions.ts

```diff
@@ -595,6 +642,71 @@ class Client implements v1.DockerDesktopClient {
         };
       });
     },
+    subscribeToEvents: (options: DockerEventSubscriptionOptions = {}, callback: DockerEventCallback): DockerEventSubscription => {
+      const eventArgs = ['events', '--format', '{{json .}}'];
+
+      if (options.filters) {
+        if (options.filters.type) {
+          options.filters.type.forEach(type => {
+            eventArgs.push('--filter', `type=${ type }`);
+          });
+        }
+        if (options.filters.event) {
+          options.filters.event.forEach(event => {
+            eventArgs.push('--filter', `event=${ event }`);
+          });
+        }
+        if (options.filters.container) {
+          options.filters.container.forEach(container => {
+            eventArgs.push('--filter', `container=${ container }`);
+          });
+        }
+        if (options.filters.label) {
+          options.filters.label.forEach(label => {
+            eventArgs.push('--filter', `label=${ label }`);
+          });
+        }
+      }
+
+      const subscriptionId = `events-${ Date.now() }-${ Math.random().toString(36).substring(2, 11) }`;
+
+      const eventProcess = this.docker.cli.exec('system', eventArgs, {
+        stream: {
+          onOutput: (data: { stdout?: string; stderr?: string }) => {
+            if (data.stdout) {
+              try {
+                const event = JSON.parse(data.stdout) as DockerEvent;
+                callback(event);
+              } catch (error) {
+                console.error('Failed to parse Docker event:', error, data.stdout);
+              }
+            }
+            if (data.stderr) {
+              console.error('Docker events stderr:', data.stderr);
+            }
+          },
+          onError: (error: any) => {
+            console.error('Docker events stream error:', error);
+          },
+          onClose: (code: number) => {
+            console.debug(`Docker events stream closed with code ${ code }`);
+          },
+          splitOutputLines: true,
+        },
+      });
+
+      eventSubscriptions.set(subscriptionId, eventProcess);
+
+      return {
+        unsubscribe: () => {
+          const proc = eventSubscriptions.get(subscriptionId);
+          if (proc) {
+            proc.close();
+            eventSubscriptions.delete(subscriptionId);
+          }
```

Comment:
It's safe to delete an entry in a map that doesn't exist:
```suggestion
          eventSubscriptions.get(subscriptionId)?.close();
          eventSubscriptions.delete(subscriptionId);
```